unyt/_array_functions.py

```diff
@@ -26,39 +26,23 @@ def array2string(a, *args, **kwargs):
     )
 
 
-def _get_conversion_factor(out, units) -> float:
-    # helper function to "product"-like functions with an 'out' argument
-    out_units = getattr(out, "units", None)
-    if out_units is None:
-        return 1.0
-    try:
-        return out_units.get_conversion_factor(units)[0]
-    except UnitConversionError as exc:
-        raise TypeError(
-            "output array is not acceptable "
-            f"(units '{out.units}' cannot be converted to '{units}')"
-        ) from exc
-
-
-@implements(np.dot)
-def dot(a, b, out=None):
+def product_helper(a, b, out, func):
```

Comment:
I think this should still be a hidden function
```suggestion
def _product_helper(a, b, out, func):
```

Reply:
It’s already in a private namespace

Reply:
good point